### PR TITLE
Union Manga: bugfix

### LINKED ISSUE #3385: Union Mangas does not upload content.
### Source information

Union Mangas 1.4.3

### Source language

Multiples (Brazilian Portuguese, Italian)

### Steps to reproduce

When opening the source it does not show the content.  An error message appears.  The content only appears in the WebView.
Screenshot_2024-06-03-10-45-09-416_app mihon
Screenshot_2024-06-03-10-45-16-514_app mihon

### Expected behavior

That the source showed the content of the site.

### Actual behavior

An error message appears and does not load content.

### Mihon/Tachiyomi version

0.16.5

### Android version

Android 12

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by choppeh**
Closes  #3385

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt [hunk 1]
```diff
@@ -1,149 +1,69 @@
...
-        try { UnionMangas.dateFormat.parse(trim())!!.time } catch (_: Exception) { 0L }
-
-    private fun ChapterDto.toChapterUrl(prefix: String) = "/${this.slugManga}/$prefix-${this.id}"
+open class Pageable<T>(
+    var currentPage: Int,
+    var totalPage: Int,
+    val data: List<T>,
+) {
+    fun hasNextPage() =
+        try { (currentPage + 1) <= totalPage } catch (_: Exception) { false }
```
Comment (AwkwardPeak7) [inline src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt]: what exception are you expceting here?

**Review by AwkwardPeak7 — COMMENTED** [review]
old urls still work?

File: src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt (hunk 1 above)
Comment (choppeh) [reply src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt]: I forgot to remove, no need to try/catch

**Comment by choppeh** [comment]
> old urls still work?

I added compatibility for the url pattern I found.

~Since Italian is no longer supported by the source, I don't really know what will happen to it.~

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**